v3/src/models/data/formula-manager.ts

```diff
@@ -347,6 +376,20 @@ export class FormulaManager {
     return displayNameMap
   }
 
+  observeDatasetChanges(dataSet: IDataSet) {
+    const dispose = onAnyAction(dataSet, mstAction => {
+      switch (mstAction.name) {
+        // When attribute is moved to a new collection, it usually affects grouping that is respected by formulas.
+        // It'd be possible to optimize this by checking formula dependencies and limit number of updates, but
+        // for now let's keep it simple and see if we encounter
```

Comment:
```suggestion
        // for now let's keep it simple and see if we encounter any problems.
```
Or end the sentence however you were originally intending to. 😉